src/Common/StringSearcher.h

```diff
@@ -811,15 +805,22 @@ class TokenSearcher : public StringSearcherBase
     size_t needle_size;
 
 public:
+
+    template <typename CharT>
+    requires (sizeof(CharT) == 1)
+    static bool isValidNeedle(const CharT * needle_, size_t needle_size_)
+    {
+        return std::none_of(needle_, needle_ + needle_size_, isTokenSeparator));
```

Comment:
Superfluous parentheses, sorry:
```suggestion
        return std::none_of(needle_, needle_ + needle_size_, isTokenSeparator);
```